src/query/functions/temporal/base.go

```diff
@@ -197,7 +202,20 @@ func (c *baseNode) Process(queryCtx *models.QueryContext, ID parser.NodeID, b bl
 		}
 	}
 
-	return c.processCompletedBlocks(processRequests, maxBlocks)
+	blocks, err := c.processCompletedBlocks(queryCtx, processRequests, maxBlocks)
+	defer closeBlocks(blocks)
+
+	if err != nil {
```

Comment:
So probably(definitely!) out of scope for this PR, but would be good to revisit temporal functions at some point and add in some logic that if any incoming block errored, it should drop any incoming blocks with the same error instead of trying to add them and starting processing before hitting it... May add an issue regarding this?

Reply:
The general function of temporal functions is that blocks come in, and are added to a cache. When enough blocks come in to fully calculate a downstream block, it is processed (which can error).  At the moment, if the processing fails, that particular request will fail but there will still be in-flight blocks coming into the temporal function, which do not necessarily need to be computed since the entire query will error anyway

Reply:
Ah ok cool, thanks for the clarification. That seems like a kind of generic problem--other parts of the query should be cancelled once one part fails. Do we have any general cancellation mechanism already? I suppose we could check the context periodically during each node's `Process`; that would probably be the idiomatic way to do it. No idea how often "periodically" should be though haha.